modules/test/playwright/tests/document-library-web/dmFolder.spec.ts

```diff
@@ -38,3 +50,92 @@ test(
 		await page.goto('/en');
 	}
 );
+
+test(
+	'Test Advance Update permission for DM folder',
+	{tag: '@LPD-46006'},
+	async ({
+		apiHelpers,
+		documentLibraryEditFolderPage,
+		documentLibraryPage,
+		page,
+		site,
+	}) => {
+		const testUser = await apiHelpers.headlessAdminUser.postUserAccount();
+
+		const role =
+			await apiHelpers.headlessAdminUser.getRoleByName('Site Member');
+
+		await apiHelpers.headlessAdminUser.assignUserToSite(
+			String(role.id),
+			site.id,
+			testUser.id
+		);
+
+		await documentLibraryPage.goto(site.friendlyUrlPath);
+		await documentLibraryPage.goToCreateNewFolder();
+
+		const title = getRandomString();
+
+		await documentLibraryEditFolderPage.fillTitle(title);
+
+		const checkboxUpdate = page.getByLabel(
+			'Give Update permission to users with role Site Member.'
+		);
+		if (await checkboxUpdate.isChecked()) {
+			await checkboxUpdate.uncheck();
+		}
+
+		const checkboxAdvanceUpdate = page.getByLabel(
+			'Give Advance Update permission to users with role Site Member.'
+		);
+		if (!(await checkboxAdvanceUpdate.isChecked())) {
+			await checkboxAdvanceUpdate.check();
+		}
+
+		await page.getByRole('button', {name: 'Save'}).click();
+
+		await documentLibraryPage.goToEditFolder(title);
+
+		await documentLibraryEditFolderPage.title.waitFor();
+
+		const currentURL = page.url();
```

Comment:
Check if we can use [`page.waitForURL`](https://playwright.dev/docs/api/class-page#page-wait-for-url) for a specific URL instead of waiting for UI elements.